### ISSUE TITLE
After Flatten, pre-loaded VGG net model shape is (?, ?)

### ISSUE THREAD

**Issue by ghost** [keras-team/keras#10112]
I preloaded the VGG net model in `keras.applications`. 

    from keras import applications, regularizers, Model
    from keras.layers import Dropout, Flatten, Dense
    from keras.constraints import max_norm
    l2_lambda = 0.01
    img_width, img_height, img_channels = 224, 224, 3
    
    base_model = applications.VGG19(
        weights="imagenet",
        include_top=False,
        input_shape=(img_width, img_height, img_channels))
    
    x = base_model.output
    print(x.shape)
    x = Flatten()(x)
    print(x.shape)

This print `(?, 7, 7, 512)` followed by `(?, ?)`. Shouldnt the latter be `(?, 25088)`?

**Comment by Dref360** [comment]
? just means that we do not know the shape.

in Flatten, we use a reshape with [batch_size, -1]. The backends don't compute the true value of -1.

**Comment by mobiusinversion** [comment]
@Dref360 why would the reshape be with [batch_size, -1], when the shape is trivial to compute by multiplying the input dims?  I would suggest this should be a feature request...

**Comment by Dref360** [comment]
It wouldn't work for most use-cases.
If you have one unknown dimension you can't compute the shape.
Ex : Fully-convolutional NN, most RNN, Sequences of all sort.
